docs/jumpstart/week-6.md

```diff
@@ -0,0 +1,697 @@
+  // And we can make sure deletePost was called the same way.
+  Mockito.verify(mockPostTable, Mockito.times(2)).deletePost(Mockito.anyInt());
+  Mockito.verify(mockPostTable).deletePost(5);
+}
+```
+
+## New Request Types
+
+Last week, we learned about the `GET` request type. There are a lot of other common request types too.
+
+### Common Request Types
+
+The most common request types (other than `GET`) are `POST`, `PUT`, `PATCH`, and `DELETE`. Of these, the one you 
+probably see the most is `POST`. Every time you send a form by hitting a submit button, like when entering credit card 
+information, typing in an address, or signing up for something, a `POST` request is sent with your data in the request 
+body to the server. The other types aren't as common, but you've probably used them all a bunch without realizing it! 
+When thinking in terms of CRUD (create, read, update, delete), this is what those types represent:
+
+| Method | CRUD type | What it does |
+| :---   | :---      | :-----       |
+| GET    | Read      | Only _get_ information |
+| POST   | Create    | Create new information |
+| PUT    | Replace (Update) | Replace currently existing information |
+| PATCH  | Modify (Update) | Modify currently existing information |
+| DELETE | Delete | Delete information |
+
+Requests can also have a request body (`GET` requests usually have all relevant information in its headers or path).
+It's common that request bodies are in JSON format, but there are other formats that can be used too. `GET` requests 
+are also the requests that are used in browsers when typing in a URL, while the other types may be used 
+when you submit a form/log in or perform other operations. Because of this, `POST`, `PUT`, `PATCH`, and `DELETE` methods
+are more closely related to each other than to `GET` requests.
+
+See the [comparison between GET and POST requests](https://www.w3schools.com/tags/ref_httpmethods.asp).
+
+On the backend side, to create a route for one of those other method types, you just substitute `router.get()` for
+the request method you want to use. 
+
+Examples:
+```java 
+Route route = router.post("/post/route");
+```
+```java 
+Route route = router.put("/put/route");
+```
+```java 
+Route route = router.patch("/patch/route");
+```
+```java 
+Route route = router.delete("/delete/route");
+```
+
+### Other Request Types
+
+There are many other request types that are used too. In practice, though, you'll probably never see them. The most 
+commonly used requests are `GET` and `POST`, followed by `PUT`, `PATCH`, and `DELETE`.
+MDN has a list of [other request types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods) if you're interested.
+
+## DTOs and Marshalling/Unmarshalling
+
+### What are DTOs
+
+DTOs, or Data Transfer Objects, are objects whose only purpose is to contain and transfer data around. They usually
+only have methods for getting and setting data. Sometimes serialization and deserialization is included, but we'll 
```

Comment:
This is just an idea but this is already long so I don't know if we want to extend it but maybe explain what serialization and deserialization is?

Reply:
I threw a footnote in